File: src/MyServiceBus.Abstractions/RetryConfigurator.cs
using System;

namespace MyServiceBus;

public class RetryConfigurator
{
    public int RetryCount { get; private set; }
    public TimeSpan? Delay { get; private set; }

    public void Immediate(int retryCount)
    {
        RetryCount = retryCount;
        Delay = null;
    }

    public void Interval(int retryCount, TimeSpan interval)
    {
        RetryCount = retryCount;
        Delay = interval;
    }
}

File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/RetryConfigurator.java
package com.myservicebus;

import java.time.Duration;

public class RetryConfigurator {
    private int retryCount;
    private Duration delay;

    public void immediate(int retryCount) {
        this.retryCount = retryCount;
        this.delay = null;
    }

    public void interval(int retryCount, Duration delay) {
        this.retryCount = retryCount;
        this.delay = delay;
    }

    public int getRetryCount() {
        return retryCount;
    }

    public Duration getDelay() {
        return delay;
    }
}
